pkgs/applications/networking/instant-messengers/ricochet/default.nix

```diff
@@ -0,0 +1,44 @@
+{ stdenv, fetchurl, pkgconfig
+, qtbase, qttools, makeQtWrapper
+, qtmultimedia, qtquick1, qtquickcontrols
+, openssl, protobuf
+}:
+
+stdenv.mkDerivation rec {
+  name = "ricochet-${version}";
+  version = "1.1.1";
+
+  src = fetchurl {
+    url = "https://github.com/ricochet-im/ricochet/archive/v${version}.tar.gz";
+    sha256 = "0y79igzgl9xn00981zcyxdlks7vnqxhb4rq3x8gwxm0yr98p39ms";
+  };
+
+  buildInputs =
+    [
+      qtbase qttools qtmultimedia qtquick1 qtquickcontrols openssl protobuf
+    ];
+
+  nativeBuildInputs = [ pkgconfig makeQtWrapper ];
+
+  preConfigure = ''
+    export NIX_CFLAGS_COMPILE="$NIX_CFLAGS_COMPILE $(pkg-config --cflags openssl)"
+  '';
+
+  configurePhase = ''
```

Comment:
Instead of setting `configurePhase`, I would recommend doing this instead:
```
configureScript = "qmake";
dontAddPrefix = true;
configureFlags = [ "PREFIX=$(out)" "DEFINES+=RICOCHET_NO_PORTABLE" ];
```
This way, you don't need to worry about invoking the `preConfigure` hook. Setting the `PREFIX` should also make it the `mkdir ...; cp ...` steps in `installPhase` unnecessary, but you should double-check that by building it yourself because every QMake project is a little different.